parachains/runtimes/collectives/collectives-polkadot/src/fellowship/migration.rs

```diff
@@ -0,0 +1,260 @@
+				);
+				return weight
+			}
+
+			let mut max_rank = 0;
+			for (rank, account_id32) in FellowshipAddresses::get() {
+				let who: T::AccountId = account_id32.into();
+				Members::<T, I>::insert(&who, MemberRecord::new(rank));
+				weight.saturating_accrue(T::DbWeight::get().writes(1));
+				for inner_rank in 0..rank + 1 {
+					let index = MemberCount::<T, I>::get(rank);
+					MemberCount::<T, I>::insert(inner_rank, index + 1);
+					IdToIndex::<T, I>::insert(inner_rank, &who, index);
+					IndexToId::<T, I>::insert(inner_rank, index, &who);
+					max_rank = max_rank.max(inner_rank);
+					// 2 writes to IdToIndex and IndexToId.
+					weight.saturating_accrue(T::DbWeight::get().writes(2));
+				}
+			}
+			// writes to MemberCount.
+			weight.saturating_accrue(T::DbWeight::get().writes((max_rank as u64) + 1));
+
+			weight
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(state: Vec<u8>) -> Result<(), &'static str> {
+			assert_eq!(MemberCount::<T, I>::get(0), 45, "invalid members count at rank 0.");
+			assert_eq!(MemberCount::<T, I>::get(1), 45, "invalid members count at rank 1.");
+			assert_eq!(MemberCount::<T, I>::get(2), 24, "invalid members count at rank 2.");
+			assert_eq!(MemberCount::<T, I>::get(3), 17, "invalid members count at rank 3.");
+			assert_eq!(MemberCount::<T, I>::get(4), 10, "invalid members count at rank 4.");
+			assert_eq!(MemberCount::<T, I>::get(5), 7, "invalid members count at rank 5.");
+			assert_eq!(MemberCount::<T, I>::get(6), 3, "invalid members count at rank 6.");
+			assert_eq!(MemberCount::<T, I>::get(7), 0, "invalid members count at rank 7.");
+			Ok(())
+		}
+	}
+}
+
+#[cfg(test)]
+pub mod tests {
+	use super::import_kusama_fellowship::FellowshipAddresses;
+	use crate::{FellowshipCollectiveInstance as Fellowship, Runtime, System};
+	use frame_support::traits::OnRuntimeUpgrade;
+	use pallet_ranked_collective::Rank;
+	use parachains_common::AccountId;
+	use sp_core::crypto::Ss58Codec;
+	use sp_runtime::AccountId32;
+
+	#[test]
+	fn check_fellowship_addresses() {
+		let fellowship_addresses = FellowshipAddresses::get();
+		let kusama_fellowship_ss58: [(Rank, _); 45] = [
+			(6, "FcxNWVy5RESDsErjwyZmPCW6Z8Y3fbfLzmou34YZTrbcraL"),
+			(6, "J8ww78Qx3LVLW54bva3t4SzXcWKMdUWHEZR3V2VNKbmQgE8"),
+			(6, "FFFF3gBSSDFSvK2HBq4qgLH75DHqXWPHeCnR1BSksAMacBs"),
+			(5, "G7YVCdxZb8JLpAm9WMnJdNuojNT84AzU62zmvx5P1FMNtg2"),
+			(5, "FczL661VujsMFqZT32Y3iBmNpSxUw4gMnq2JtdoFT6rJzr4"),
+			(5, "Dcm1BqR4N7nHuV43TXdET7pNibt1Nzm42FggPHpxKRven53"),
+			(5, "J9nD3s7zssCX7bion1xctAF6xcVexcpy2uwy4jTm9JL8yuK"),
+			(4, "EGVQCe73TpFyAZx5uKfE1222XfkT3BSKozjgcqzLBnc5eYo"),
+			(4, "HL8bEp8YicBdrUmJocCAWVLKUaR2dd1y6jnD934pbre3un1"),
+			(4, "HyBryanRsB1GGKa9ZfqvRc3XpTDipYyRvxNNyZYfWFcenhd"),
+			(3, "GAWwkmjbLhM5pnAVbdZEkwd3QjHE7kaxJSCi3Ec91Q3QSDW"),
+			(3, "H25aCspunTUqAt4D1gC776vKZ8FX3MvQJ3Jde6qDXPQaFxk"),
+			(3, "GtLQoW4ZqcjExMPq6qB22bYc6NaX1yMzRuGWpSRiHqnzRb9"),
+			(3, "G5MVrgFmBaYei8N6t6DnDrb8JE53wKDkazLv5f46wVpi14y"),
```

Comment:
https://kusama.subscan.io/extrinsic/16876631-2

```suggestion
			(3, "15db5ksZgmhWE9U8MDq4wLKUdFivLVBybztWV8nmaJvv3NU1"),
```